integration/run/run_test.go

```diff
@@ -1036,16 +1037,28 @@ func TestUsingComputeClasses(t *testing.T) {
 
 	for _, tt := range checks {
 		asClusterComputeClass := adminv1.ClusterComputeClassInstance(tt.computeClass)
-		for kind, computeClass := range map[string]runtimeclient.Object{"projectcomputeclass": &tt.computeClass, "clustercomputeclass": &asClusterComputeClass} {
-			t.Run(fmt.Sprintf("%v_%v", kind, tt.name), func(t *testing.T) {
+		// Perform the same test cases on both Project and Cluster ComputeClasses
+		for kind, computeClass := range map[string]crClient.Object{"projectcomputeclass": &tt.computeClass, "clustercomputeclass": &asClusterComputeClass} {
+			testcase := fmt.Sprintf("%v-%v", kind, tt.name)
+			t.Run(testcase, func(t *testing.T) {
 				if !tt.noComputeClass {
 					if err := kclient.Create(ctx, computeClass); err != nil {
 						t.Fatal(err)
 					}
+
+					// Clean-up and gurantee the computeclass doesn't exist after this test run
 					t.Cleanup(func() {
 						if err = kclient.Delete(context.Background(), computeClass); err != nil && !apierrors.IsNotFound(err) {
 							t.Fatal(err)
 						}
+						err := helper.EnsureDoesNotExist(ctx, func() (crClient.Object, error) {
```

Comment:
I don't understand why this is necessary. If we delete it and that delete is successful, why do we have to ensure it doesn't exist?

Reply:
I'm trying to find a good source for this that backs me up here, but if I remember correctly deletes are asynchronous.  and thus deletes through the client are asynchronous. So the call to delete passes but it sticks around as things like network calls, finalizers, etc. finish up. The suite is failing because either apps or ComputeClasses are not getting deleted in time between test cases and collisions occur.

Reply:
It could also be that the resource exists in the cache still even though it has been deleted.

Reply:
For context - @thedadams and I talked about this offline. We tried getting around using the `EnsureDoesNotExist` function using a `PropogationPolicy` of `foreground` to no avail in the delete to no avail

Reply:
Talked offline further - this approach is fine given the circumstances.